statsproduct.php

```diff
@@ -444,32 +444,34 @@ protected function getData($layers)
         } else {
             $product = new Product($this->id_product, false, (int)$this->getLang());
 
-            $comb_array = array();
-            $assoc_names = array();
-            $combinations = $product->getAttributeCombinations((int)$this->getLang());
-            foreach ($combinations as $combination) {
-                $comb_array[$combination['id_product_attribute']][] = array(
-                    'group' => $combination['group_name'],
-                    'attr' => $combination['attribute_name']
-                );
-            }
-            foreach ($comb_array as $id_product_attribute => $product_attribute) {
-                $list = '';
-                foreach ($product_attribute as $attribute) {
-                    $list .= trim($attribute['group']).' - '.trim($attribute['attr']).', ';
-                }
-                $list = rtrim($list, ', ');
-                $assoc_names[$id_product_attribute] = $list;
-            }
-
             $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($this->query);
             foreach ($result as $row) {
                 $this->_values[] = $row['total'];
-                $this->_legend[] = @$assoc_names[$row['product_attribute_id']];
+                $this->_legend[] = $this->removeProductNameInCombinationName($product->name, $row['product_name']);
             }
         }
     }
 
+    /**
+     * @param string $productName
+     * @param string $combinationName
+     *
+     * @return string
+     */
+    protected function removeProductNameInCombinationName(string $productName, string $combinationName): string
+    {
+        $combinationFullNameSepatators = ['-', ':'];
+        $combinationName = str_replace($productName, '', $combinationName);
+        $combinationName = trim($combinationName);
+
+        if(in_array($combinationName[0], $combinationFullNameSepatators)){
```

Comment:
So, if it starts with `-:` the ltrim is only did once, it's the behavior?
Could you please improve the description to better understand your need?

Reply:
Can you please check https://github.com/PrestaShop/PrestaShop/issues/25049